dor/providers/process_basic_image.py

```diff
@@ -398,6 +402,144 @@ def run(self) -> None:
         )
 
 
+@dataclass
+class ExtractImageTextCoordinates(Operation):
+    language: str = "eng"
+
+    def run(self) -> None:
+        file_info = FileInfo(
+            identifier=self.accumulator.file_set_identifier.identifier,
+            basename=self.accumulator.file_set_identifier.basename,
+            uses=[UseFunction.service, UseFormat.text_coordinates],
+            mimetype=Mimetype.XML.value,
+        )
+
+        service_result_file = self.accumulator.get_file(
+            function=[UseFunction.service], format=UseFormat.image
+        )
+
+        alto_xml = ImageTextExtractor(image_path=service_result_file.file_path, language=self.language).alto
+        alto_file_path = self.accumulator.file_set_directory / file_info.path
+        alto_file_path.write_text(alto_xml)
+
+        try:
+            tech_metadata = TechnicalMetadata.create(alto_file_path)
+        except JHOVEDocError:
+            return None
+
+        event = create_preservation_event(
+            "extract text coordinates with OCR", self.accumulator.collection_manager_email
+        )
+
+        self.accumulator.add_file(
+            ResultFile(
+                file_path=alto_file_path,
+                tech_metadata=tech_metadata,
+                file_info=file_info,
+                source_file_result=service_result_file,
+                event=event,
+            )
+        )
+
+
+@dataclass
+class ExtractImageText(Operation):
+    language: str = "eng"
+
+    def run(self) -> None:
+        file_info = FileInfo(
+            identifier=self.accumulator.file_set_identifier.identifier,
+            basename=self.accumulator.file_set_identifier.basename,
+            uses=[UseFunction.service, UseFormat.text_plain],
+            mimetype=Mimetype.TXT.value
```

Comment:
This was a little awkward. I added a vanilla `Mimetype.TXT` because we don't actually know the character set of the output -- and it doesn't really matter for `FileInfo`.

Reply:
The dilemma is using `file_info.path` before we get the technical metadata?

Reply:
Yeah, basically. We can't get the mimetype from the technical metadata because the file needs to be written before JHOVE can do anything. We could rewrite the file name, but we're not including character set info in the file name.

Reply:
We should leave a comment here as a signpost/TODO in case we revisit this before release. (I'm content to move on but not to leave this because of the dataclass initialization.)

Reply:
I'm not sure I understand. I don't think it's related to the dataclass initialization... What would you add as a comment?

Reply:
Okay, done. https://github.com/mlibrary/dor-py/pull/76/commits/0277eff78720194364c0d8ee99e36d1d2b702b80